tools/old_config_maker/graph_node.py

```diff
@@ -0,0 +1,260 @@
+
+        Button(master, text="Add Metric", command=lambda: close_config_window(self, master, entries)).grid(row=9, column=0)
+
+        master.mainloop()
+
+    def delete(self):
+        """
+        Deletes all tkinter items in node.
+
+        Returns
+        -------
+        Self object so it can be deleted with all of its stuff in one line
+        """
+
+        for value in self.items.values():
+            if type(value) is dict:
+                for item in value.values():
+                    item.destroy()
+            else:
+                value.destroy()
+
+        return self
+
+    def set_grid(self, row_offset=None):
+        """
+        Every button/item has a grid based location and this places every item to that grid space
+
+        Parameters
+        ----------
+        row_offset: number, default=None
+            How many rows this graph_node should start down from top of frame
+        """
+
+        if row_offset: self.row_offset = row_offset
+
+        rolling_offset = self.row_offset  # If checkboxes take extra lines, the lines underneath will drop one
+
+        for key, value in self.items.items():
+            if key in self.item_locations.keys():
+                grid_values = self.item_locations[key]
+
+                if type(value) is dict:
+                    for i, key in enumerate(value.keys()):
+                        value[key].grid(sticky="W", column=i % GraphNode.checkbox_width,
+                                        row=rolling_offset + grid_values + int(i / GraphNode.checkbox_width))
+
+                    rolling_offset += int(len(value) / GraphNode.checkbox_width)
+                else:
+                    value.grid(column=grid_values[0],
+                               row=grid_values[1] + rolling_offset if grid_values[1] > 1 else self.row_offset,
+                               columnspan=grid_values[2] if len(grid_values) > 2 else 1)
+
+        self.height = rolling_offset + GraphNode.rows_per_graph
+
+    def update_title(self):
+        """
+        Allows you to update the name of the graph
+        """
+
+        if isinstance(self.items['title'], Entry):
+            name = self.items['title'].get()
+
+            self.items['title'].destroy()
+
+            self.items['title'] = Label(self.tab, text=name, font=("Arial Bold", 15), bg="#66AA33", borderwidth=1)
+
+            self.items['update_title']['text'] = "Change Name"
+
+        else:
+            self.items['title'].destroy()
+
+            self.items['title'] = Entry(self.tab, width=20)
+
+            self.items['update_title']['text'] = "Submit"
+
+        self.set_grid()
+
+    # TODO - NOT FOR BASE WORKING
+    def reset(self, to_reset=None):
+        # TODO - Add default item settings
+        if type(to_reset) is not list: to_reset = [to_reset]
+        if not to_reset: to_reset = [key for key in self.items.keys()]
+
+        for item in to_reset:
+            if item == 'check_boxes':
```

Comment:
'check_boxes' constant is used multiple times, consider making it a constant or having an enum for different control labels.